src/parquet/column_reader.cc

```diff
@@ -212,6 +121,9 @@ bool ColumnReader::ReadNewPage() {
       // Read a data page.
       num_buffered_values_ = current_page_header_.data_page_header.num_values;
```

Comment:
Yes -- after decoding the definition levels you can count the number of materialized values and then request that many to be decoded in batch. 

For applications that are looking to save as many CPU cycles as possible we can figure out an API that avoids this; for now I'd like to get to a correct-and-tested state and then focus on optimizations.